File: app/src/main/java/com/sudoajay/pdf_viewer/HelperClass/FileSize.java
package com.sudoajay.pdf_viewer.HelperClass;

import java.text.DecimalFormat;

public class FileSize {

    public static String Convert_It(long size) {
        if (size > (1024 * 1024 * 1024)) {
            // GB
            return GetDecimal2Round((double) size / (1024 * 1024 * 1024)) + " GB";
        } else if (size > (1024 * 1024)) {
            // MB
            return GetDecimal2Round((double) size / (1024 * 1024)) + " MB";

        } else {
            // KB
            return GetDecimal2Round((double) size / (1024)) + " KB";
        }

    }


    private static String GetDecimal2Round(double time){
        DecimalFormat df = new DecimalFormat("#.#");
        return Double.valueOf(df.format(time)).toString();
    }
}

File: app/src/main/java/com/sudoajay/pdf_viewer/helperClass/FileSize.kt
package com.sudoajay.pdf_viewer.helperClass

import java.text.DecimalFormat

object FileSize {
    @JvmStatic
    fun convertIt(size: Long): String {
        return when {
            size > 1024 * 1024 * 1024 -> { // GB
                getDecimal2Round(size.toDouble() / (1024 * 1024 * 1024)) + " GB"
            }
            size > 1024 * 1024 -> { // MB
                getDecimal2Round(size.toDouble() / (1024 * 1024)) + " MB"
            }
            else -> { // KB
                getDecimal2Round(size.toDouble() / 1024) + " KB"
            }
        }
    }

    private fun getDecimal2Round(time: Double): String {
        val df = DecimalFormat("#.#")
        return java.lang.Double.valueOf(df.format(time)).toString()
    }
}